### PR TITLE
Don't create subtasks when not using tools

### PR DESCRIPTION
**Pull request by collindutter**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
Don't create subtasks when not using tools 
## Issue ticket number and link
Introduced via https://github.com/griptape-ai/griptape/pull/1463

### CODE REVIEW

File: griptape/tasks/prompt_task.py [hunk 1]
```diff
@@ -203,21 +203,24 @@ def try_run(self) -> BaseArtifact:
...
+                        result = self.prompt_driver.run(self.prompt_stack)
+                        subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
                 else:
-                    subtask.run()
-
-                    result = self.prompt_driver.run(self.prompt_stack)
-                    subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
-            else:
-                break
+                    break
```
Comment (collindutter) [inline griptape/tasks/prompt_task.py:219]: Not introduced by this change

File: griptape/tasks/prompt_task.py [hunk 2]
```diff
@@ -203,21 +203,24 @@ def try_run(self) -> BaseArtifact:
...
-        subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
-
-        while True:
-            if subtask.output is None:
-                if len(self.subtasks) >= self.max_subtasks:
-                    subtask.output = ErrorArtifact(f"Exceeded tool limit of {self.max_subtasks} subtasks per task")
+        if self.tools:
+            subtask = self.add_subtask(ActionsSubtask(result.to_artifact()))
+
+            while True:
```
Comment (dylanholmes) [inline griptape/tasks/prompt_task.py:209]: You could avoid some nesting if you flip the condition:

```py
if not self.tools:
    self.output = result.to_artifact()
    return self.output

while True:
    ...
```

**Review by dylanholmes — APPROVED** [review]
Optional suggestion

File: griptape/tasks/prompt_task.py (hunk 2 above)
Comment (collindutter) [reply griptape/tasks/prompt_task.py:209]: I'd rather keep the conditional positive and solve the ugly nested code separately [here](https://github.com/griptape-ai/griptape/issues/1472).

**Pull request merged**